apm-sniffer/apm-sdk-plugin/thrift-plugin/src/main/java/org/apache/skywalking/apm/plugin/thrift/wrapper/ServerInProtocolWrapper.java

```diff
@@ -45,6 +45,7 @@
     private static final ILog LOGGER = LogManager.getLogger(ServerInProtocolWrapper.class);
     private static final StringTag TAG_ARGS = new StringTag("args");
     private AbstractContext context;
+    private static ThreadLocal<Boolean> HAVE_CREATED = ThreadLocal.withInitial(() -> Boolean.FALSE);
```

Comment:
TracingContext has native running context for this case, could you consider to use that? It could be auto cleared when the tracing context finished.

Reply:
It seems like the life cycle of a TraceContext is exactly one trace. Thinking of such a circumstance that we have A, B, C three services and they are inter-communicated through RPC like A->B->C, with both A and C mounted the Agent while B not. Then, in the above situation, C will start a new trace since the trace from A is stopped at B. Hence, C could not obsever A with the use of TraceContext.

Reply:
How is your description relating to what I asked? I am talking about don't your own threadlocal, use the one core provided. Do you misunderstand somehow?

Reply:
SkyWalking provides a safe `RuntimeContext` which is alive during a tracing. So we prefer to use this rather than directly ThreadLocal. 
More details see `ContextManager#getRuntimeContext()`, please.

Reply:
Sorry,  I misunderstand in some aspect.
Thanks for your advice.